src/components/GlobalFilter/tagsApi.ts

```diff
@@ -70,38 +73,109 @@ export function getAllTags({ search, activeTags, registeredWith }: TagFilterOpti
     // @ts-ignore
     registeredWith ? registeredWith : undefined,
     undefined,
-    {
-      query: generateFilter(buildFilter(workloads, SID)),
-    }
+    { query: generateFilter(buildFilter(workloads, SID)) }
   );
+
+  // @ts-ignore
+  chromeStore.set(tagsAtom, (prev) => ({
+    ...prev,
+    isLoaded: true,
+    // Update pagination state
+    page: pagination?.page || 1,
+    perPage: pagination?.perPage || 10,
+    // Transform API results into the required GlobalFilterTag[] structure for TagsModal
+    items: [
+      {
+        id: 'tags-group',
+        name: 'Tags',
+        tags: ((response as any).data?.results || (response as any).results || [])
+          .filter((result: any) => result?.tag?.key && typeof result.tag.key === 'string' && result.tag.key.trim() !== '')
+          .map((result: any) => {
+            const namespace = result.tag.namespace || '';
+            const key = result.tag.key || '';
+            const value = result.tag.value || '';
+            return {
+              tag: {
+                id: `${namespace}/${key}=${value}`,
+                key: key,
+                value: value,
+                namespace: namespace,
+              },
+              count: result.count || 0,
+            };
+          })
+          .filter((item: any) => item.tag.key && item.tag.id), // Additional safety check
+      },
+    ],
+    // @ts-ignore
+    count: (response as any).data?.count || (response as any).count,
+    // @ts-ignore
+    total: (response as any).data?.total || (response as any).total,
+  }));
 }
 
-export function getAllSIDs({ search, activeTags, registeredWith }: TagFilterOptions = {}, pagination: TagPagination = {}) {
+export async function getAllSIDs({ search, activeTags, registeredWith }: TagFilterOptions = {}, pagination: TagPagination = {}) {
   const [workloads, SID, selectedTags] = flatTags(activeTags, false, true);
 
-  return sap.apiSystemProfileGetSapSids(
+  const response = await sap.apiSystemProfileGetSapSids(
     search,
-    selectedTags, // tags
+    selectedTags,
     (pagination && pagination.perPage) || 10,
     (pagination && pagination.page) || 1,
-    undefined, // staleness,
+    undefined,
     // @ts-ignore
     registeredWith ? registeredWith : undefined,
     undefined,
     {
       query: generateFilter(buildFilter(workloads, SID)),
     }
   );
+
+  // @ts-ignore
```

Comment:
Same here, do not use this annotation if you don't have and cast the response variable, so you don't have to constantly cast it to any.